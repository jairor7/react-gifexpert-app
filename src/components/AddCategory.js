import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ categories, setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };
  const handleSubmit = (event) => {
    event.preventDefault();
    if (
      inputValue.trim() !== "" &&
      !categories.find((category) => category === inputValue)
    ) {
      setCategories((preState) => [inputValue, ...preState]);
      setInputValue("");
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
  categories: PropTypes.array.isRequired,
};

export default AddCategory;
