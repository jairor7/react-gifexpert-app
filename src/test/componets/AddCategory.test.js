import "@testing-library/jest-dom";

import AddCategory from "../../components/AddCategory";
import { shallow } from "enzyme";

describe("Pruebas de funcionamiento para componente <AddCategory />", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(
    <AddCategory categories={[]} setCategories={setCategories} />
  );

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(
      <AddCategory categories={[]} setCategories={setCategories} />
    );
  });

  test("Debe mostrar correctamente el componente <AddCategory />", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";
    input.simulate("change", { target: { value } });
    expect(wrapper.find("input").prop("value")).toBe(value);
  });

  test("No debe de llamar el Submit del form si la caja de texto esta vacia", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe de llamar el setCategories y limpiar la caja de texto", () => {
    const value = "Hola Mundo";
    wrapper.find("input").simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
