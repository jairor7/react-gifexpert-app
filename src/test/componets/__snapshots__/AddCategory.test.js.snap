// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de funcionamiento para componente <AddCategory /> Debe mostrar correctamente el componente <AddCategory /> 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
